meta/saisanitycheck.c

```diff
@@ -2324,26 +2338,45 @@ void check_attr_acl_field_or_action(
         META_ASSERT_FALSE(md->defaultvalue->aclaction.enable, "enable should be false");
     }
 
-    if (md->objecttype != SAI_OBJECT_TYPE_ACL_ENTRY)
+    if (md->objecttype != SAI_OBJECT_TYPE_ACL_ENTRY && md->objecttype != SAI_OBJECT_TYPE_UDF_MATCH)
     {
         META_ASSERT_FALSE(md->isaclfield, "field should be not marked as acl field");
         META_ASSERT_FALSE(md->isaclaction, "field should be not marked as acl action");
 
         return;
     }
 
-    if (md->attrid >= SAI_ACL_ENTRY_ATTR_FIELD_START &&
-            md->attrid <= SAI_ACL_ENTRY_ATTR_FIELD_END)
+    if (md->objecttype == SAI_OBJECT_TYPE_ACL_ENTRY)
     {
-        META_ASSERT_TRUE(md->isaclfield, "field should be marked as acl field");
-        META_ASSERT_FALSE(md->isaclaction, "field should be not marked as acl action");
+        if (md->attrid >= SAI_ACL_ENTRY_ATTR_FIELD_START &&
+                md->attrid <= SAI_ACL_ENTRY_ATTR_FIELD_END)
+        {
+            META_ASSERT_TRUE(md->isaclfield, "field should be marked as acl field");
+            META_ASSERT_FALSE(md->isaclaction, "field should be not marked as acl action");
+        }
+
+        if (md->attrid >= SAI_ACL_ENTRY_ATTR_ACTION_START &&
+                md->attrid <= SAI_ACL_ENTRY_ATTR_ACTION_END)
+        {
+            META_ASSERT_FALSE(md->isaclfield, "field should not be marked as acl field");
+            META_ASSERT_TRUE(md->isaclaction, "field should be marked as acl action");
+        }
     }
 
-    if (md->attrid >= SAI_ACL_ENTRY_ATTR_ACTION_START &&
-            md->attrid <= SAI_ACL_ENTRY_ATTR_ACTION_END)
+    if (md->objecttype == SAI_OBJECT_TYPE_UDF_MATCH)
     {
-        META_ASSERT_FALSE(md->isaclfield, "field should not be marked as acl field");
-        META_ASSERT_TRUE(md->isaclaction, "field should be marked as acl action");
+        if (md->attrid <= SAI_UDF_MATCH_ATTR_GRE_TYPE)
+        {
+            META_ASSERT_TRUE(md->isaclfield, "field should be marked as acl field");
+            META_ASSERT_FALSE(md->isaclaction, "field should be not marked as acl action");
+        }
+
+        if (md->attrid >= SAI_UDF_MATCH_ATTR_L4_DST_PORT_TYPE &&
+                md->attrid <= SAI_UDF_MATCH_ATTR_L4_DST_PORT_TYPE)
```

Comment:
use == sign, since you use the same attribute SAI_UDF_MATCH_ATTR_L4_DST_PORT_TYPE

Reply:
ACK.